rcl_lifecycle/include/rcl_lifecycle/rcl_lifecycle.h

```diff
@@ -26,10 +44,41 @@ extern "C"
 #include "rcl_lifecycle/default_state_machine.h"
 #include "rcl_lifecycle/visibility_control.h"
 
+/// Return a rcl_lifecycle_state_t struct with members set to `NULL` or 0.
+/**
+ * Should be called to get a null rcl_lifecycle_state_t before passing to
+ * rcl_lifecycle_state_init().
+ *
+ * \return rcl_lifecycle_state_t a initilized struct
+ */
 RCL_LIFECYCLE_PUBLIC
 rcl_lifecycle_state_t
 rcl_lifecycle_get_zero_initialized_state();
 
+/// Initialize a rcl_lifecycle_state_init.
+/**
+ * This function initializes a state based on an `id` and a `label`.
+ *
+ * The given `rcl_lifecycle_state_t` must be zero initialized with the function
+ * `rcl_lifecycle_get_zero_initialized_state()` and must not be already initialized
+ * by this function.
+ * The allocator will be used to allocate the label string.
+ *
+ * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | Yes
+ * Thread-Safe        | No
+ * Uses Atomics       | No
+ * Lock-Free          | Yes
+ *
+ * \param[inout] state pointer to the state struct to be initialized
+ * \param[in] id identifier of the state
+ * \param[in] label label of the state
+ * \param[in] allocator a valid allocator used to initialized the lifecycle state
+ * \return `RCL_RET_OK` if state was initialized successfully, or
+ * \return `RCL_RET_ERROR` if an unspecified error occurs.
```

Comment:
Through the whole code of `rcl_lifecycle.c` invalid (NULL) parameters are handled as `RCL_RET_ERROR` and these could be returned as `RCL_INVALID_ARGUMENT`, I think we could add a PR to handle this.